### ISSUE TITLE
"useless suppression of line too long" in multi-line docstring

### ISSUE THREAD

**Issue by NeilGirdhar** [PyCQA/pylint#8301]
### Bug description

```python
def f():
    """
    [Zaheer et al, 2018](https://papers.nips.cc/paper/2018/hash/90365351ccc7437a1309dc64e4db32a3-Abstract.html)
    Only positive values are allowed.
    """  # noqa, pylint: disable=line-too-long

def g():
    """
    [Zaheer et al, 2018](https://papers.nips.cc/paper/2018/hash/90365351ccc7437a1309dc64e4db32a3-Abstract.html)
    Only positive values are allowed.
    """
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
a.py:9:0: C0301: Line too long (111/100) (line-too-long)
a.py:4:0: I0021: Useless suppression of 'line-too-long' (useless-suppression)
```
Thus, pylint is neither happy with the long line nor with suppressing the error.
```


### Expected behavior

```
a.py:9:0: C0301: Line too long (111/100) (line-too-long)
```

### Pylint version

```shell
pylint 2.16.2
astroid 2.14.2
Python 3.11.1 (main, Jan  9 2023, 17:26:13) [GCC 11.3.0]
```

**Comment by mbyrnepr2** [comment]
Thanks @NeilGirdhar. I wasn't able to reproduce this one; perhaps someone else can double-check.

**Comment by NeilGirdhar** [comment]
Do you need to enable `useless-suppression`?

**Comment by mbyrnepr2** [comment]
Yes you're right! I thought it was enabled by default 🙃

**Comment by Pierre-Sassoulas** [comment]
Look like a duplicate of #7911 and more or less related to #3367

**Comment by NeilGirdhar** [comment]
@Pierre-Sassoulas Not sure it's related, but you know better than I do how the pylint code works.  Unlike #7911, it's not the disable that's pushign it over.  And unlike #3367 the disable isn't bleeding over into other lines.

The problem here seems to be that the disable's usefulness check seems to depend on whether it disabled an error on the line it's on rather that within the block of code that it applies to.  And a single disable can apply to a block of code in the case of a triple-quoted string.
